### PR TITLE
[CN-127] Add Expose Externally to Hazelcast Enterprise Operator deplo…

### PR DESCRIPTION
**Pull request by leszko**
…yment

### CODE REVIEW

**Review by Serdaro — APPROVED** [review]

File: docs/modules/deploy/pages/deploying-with-operator.adoc [hunk 1]
```diff
@@ -95,6 +95,43 @@ Members {size:3, ver:3} [
...
+
+Note that you can configure the `expose-externally` feature with the following parameters:
+
+* `type`: the way Hazelcast members are exposed, which implies Hazelcast client type (Unisocket or Smart)
+** `Unisocket`: all members are exposed via one Kubernetes service
+** `Smart`: each Hazelcast member is exposed with a separate Kubernetes service
+* `discoveryServiceType`: Type of Kubernetes service used for discovering Hazelcast cluster
+** `LoadBalancer` (default): Hazelcast client configuration `<load-balancer-external-ip>:5701`
+** `NodePort`: Hazelcast client configuration `<node-address>:<node-port>`
+* `memberAccess`: the way how each Hazelcast member is accessed
```
Comment (JakeSCahill) [inline docs/modules/deploy/pages/deploying-with-operator.adoc:130]:
```suggestion
* `memberAccess`: The way in which clients access Hazelcast members.
```

**Review by JakeSCahill — CHANGES_REQUESTED** [review]

File: docs/modules/deploy/pages/deploying-with-operator.adoc [hunk 2]
```diff
@@ -95,6 +95,43 @@ Members {size:3, ver:3} [
...
+** `NodePortExternalIP` (default): Hazelcast client is outside the Kubernetes network, so members need to be accessed via the public IP assigned to each Kubernetes Node
+** `NodePortNodeName`: Hazelcast client is inside the same network as the Kubernetes nodes, so members can access Kubernetes nodes by their hostnames
```
[the remark this answers is not in the document]
Comment (leszko) [reply docs/modules/deploy/pages/deploying-with-operator.adoc:132]: yes

File: docs/modules/deploy/pages/deploying-with-operator.adoc [hunk 3]
```diff
@@ -95,6 +95,43 @@ Members {size:3, ver:3} [
...
+
+* `type`: The way in which Hazelcast members are exposed to clients.
+** `Unisocket`: Members are exposed via one Kubernetes service.
+** `Smart`: Each member is exposed via a separate Kubernetes service.
+* `discoveryServiceType`: Type of Kubernetes service used for discovering members in a Hazelcast cluster.
+** `LoadBalancer` (default): Hazelcast client configuration `<load-balancer-external-ip>:5701`.
+** `NodePort`: Hazelcast client configuration `<node-address>:<node-port>`.
+* `memberAccess`: The way in which clients access Hazelcast members.
+** `NodePortExternalIP` (default): Hazelcast clients are outside the Kubernetes network, so a client can access members only via the public IP address that's assigned to their Kubernetes nodes.
+** `NodePortNodeName`: Hazelcast client is inside the same network as the Kubernetes nodes, so members can access Kubernetes nodes by their hostnames
```
Comment (JakeSCahill) [inline docs/modules/deploy/pages/deploying-with-operator.adoc:132]:
```suggestion
** `NodePortNodeName`: Hazelcast clients are inside the same network as the Kubernetes nodes, so clients can access a member by the hostname of its Kubernetes node.
```

**Review by JakeSCahill — APPROVED** [review]
Looks good, thanks @leszko 

I'd just clarify the point about addressing members by their Kubernetes hostnames.

**Pull request merged**